lets-encrypt/genarate_cert.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+# Methods
+# https://certbot.eff.org/docs/using.html#certbot-command-line-options
+standalone() {
+# Generate certs
+if eval "certbot certonly --standalone --pre-hook 'service apache2 stop' --post-hook 'service apache2 start' $default_le"
+then
+    return 0
+else
+    return 1
+fi  
+}
+
+tls-alpn-01() {
+if eval "certbot certonly --preferred-challenges tls-alpn-01 $default_le"
+then
+    return 0
+else
+    return 1
+fi  
+}
+
+dns() {
+if eval "certbot certonly --manual --manual-public-ip-logging-ok --preferred-challenges dns $default_le"
+then
+    return 0
+else
+    return 1
+fi  
+}   
+
+methods=(standalone dns)
+
+create_config() {
+#Check if $CERTFILES exists
+if [ -d "$CERTFILES" ]
+ then
+    # Generate DHparams chifer
+    if [ ! -f "$DHPARAMS" ]
+    then
+        openssl dhparam -dsaparam -out "$DHPARAMS" 4096
+    fi
+    # Activate new config
+    if [ "$2" == "nextcloud" ]
+    then
+        check_command bash "$SCRIPTS/test-new-config.sh" "$1.conf"
+    else
+        printf "%b" "${IGreen}Certs are generated!\n${Color_Off}"
+        a2ensite "$1.conf"
+        restart_webserver
+        # Install OnlyOffice
+        occ_command app:install $2
+        return 0
+    fi
+fi  
+}   
+    
+attempts_left() {
+if [ "$1" == "standalone" ]
+then
+    printf "%b" "${ICyan}It seems like no certs were generated, we will do 1 more try.\n${Color_Off}"
+    any_key "Press any key to continue..."
+#elif [ "$method" == "tls-alpn-01" ]
+#then
+#    printf "%b" "${ICyan}It seems like no certs were generated, we will do 1 more try.\n${Color_Off}"
+#    any_key "Press any key to continue..."
+elif [ "$1" == "dns" ]
+then
+    printf "%b" "${IRed}It seems like no certs were generated, please check your DNS and try again.\n${Color_Off}"
```

Comment:
use `print_text_in_color` instead. Please change in the original script as well if you find any occurrences like this.

Reply:
> use `print_text_in_color` instead. Please change in the original script as well if you find any occurrences like this.

I am sorry，most of code is copied from original scripts 。I will do it as your comments when I complete all work， I will notify you 。